specifications_dbus.md

```diff
@@ -0,0 +1,92 @@
+# Specifications
+
+## Index
+
+* [Connection](#connection)
+* [Connector](#connector)
+* [Connector API](#connector-API)
+* [Distributor](#distributor)
+* [Distributor API](#distributor-api)
+* [Appendix: Implementation practices](#appendix-implementation-practices)
+
+## Connection
+
+A *connection* is created when a connector registers with a distributor. It is uniquely identified by its *token*, a random string which the connector generates as part of the registration process.
+
+## Connector
+
+A *connector* is an application, or software library used in an application, which wants to receive push messages from a UnifiedPush distributor.
+
+When a connector first wants to register, it should fetch all service names currently registered on the session bus, and filter it to only those which begin with `org.unifiedpush.Distributor.`. It should then choose one of these (see [Appendix: Implementation practices](#appendix-implementation-practices)) to use as its distributor. It must then generate a random string to use as a token, and call the [org.unifiedpush.Distributor1.Register](#orgunifiedpushdistributor1register-string--string-string) method with this token as the argument. If the return value indicates "NEW_ENDPOINT", the connector must then save the token and the endpoint received in its persistent storage. Registration is now complete.
```

Comment:
That is implementation, it can be moved into appendix

Reply:
it is org.unifiedpush.Distributor1 instead of org.unifiedpush.Distributor

Reply:
This is the name of the service, not the interface. A process using D-Bus can register a *name* for its service (in this case `org.unifiedpush.Distributor.distributorname`), and inside that service it can have *objects* (in this case an object at `/org/unifiedpush/Distributor`), and those objects can implement *interfaces* (in this case `org.unifiedpush.Distributor1`).

Reply:
ok, mb, so it is not Distributor1 *